### PR TITLE
feat: add cleanup and fix status

### LINKED ISSUE #3: support cleanup of entries marked for deletion

### PR DESCRIPTION
**Pull request by cristianmtr**
- add cleanup
- fix cleanup PSQL query
- update README
- improve /status robustness

Closes #3

### CODE REVIEW

File: executor/hnswpsql.py [hunk 1]
```diff
@@ -239,10 +239,22 @@ def status(self, **kwargs):
...
+
+        if self._kv_indexer and self._kv_indexer.initialized:
+            psql_docs = self._kv_indexer.size
+        else:
+            self.logger.warning(f'PSQL connection has not been initialized')
+
+        if self._vec_indexer:
+            hnsw_docs = self._vec_indexer.size
+            last_sync = self._vec_indexer.last_timestamp
+            last_sync = last_sync.isoformat()
```
Comment (numb3r3) [inline executor/hnswpsql.py:254]:
```suggestion
            last_sync = self._vec_indexer.last_timestamp.isoformat()
```

**Review by numb3r3 — CHANGES_REQUESTED** [review]

**Comment by numb3r3** [comment]
@cristianmtr BTW, I have another issue not related with this PR. 
I think the current syncing implementation also has the same issue as FaissPostgressIndexer https://github.com/jina-ai/executors/issues/275

File: executor/hnswpsql.py (hunk 1 above)
Comment (cristianmtr) [reply executor/hnswpsql.py:254]: too minor a change to justify the rebasing pain into the PR that depends on this.

File: README.md [hunk 2]
```diff
@@ -39,6 +39,6 @@ You can perform all the usual operations on the respective endpoints
 - `/index`. Add new data to PostgreSQL
 - `/search`. Query the HNSW index with your Documents.
 - `/update`. Update documents in PostgreSQL
-- `/delete`. Delete documents in PostgreSQL. **Note**. This only performs soft-deletion by default. This is done in order to not break the look-up of the document id after doing a search. For a hard delete, add `'soft_delete': False'` to `parameters` 
+- `/delete`. Delete documents in PostgreSQL. **Note**. This only performs soft-deletion by default. This is done in order to not break the look-up of the document id after doing a search. For a hard delete, add `'soft_delete': False'` to `parameters`. You might also perform a cleanup after a full rebuild of the HNSW index, by calling `/cleanup`.
```
Comment (maximilianwerk) [inline README.md]: One sentence per line for better git diffs :D

File: tests/integration/test_hnsw_psql.py [hunk 3]
```diff
@@ -138,3 +137,36 @@ def test_replicas_integration(docker_compose, get_documents):
...
+        # default to soft delete
+        f.delete(docs)
+        result = f.post('/status', None, return_results=True)
+        result_docs = result[0].docs
+        assert int(result_docs[0].tags['psql_docs']) == len(docs)
+
+        f.post(on='/cleanup')
+        result = f.post('/status', None, return_results=True)
+        result_docs = result[0].docs
+        assert int(result_docs[0].tags['psql_docs']) == 0
```
Comment (maximilianwerk) [inline tests/integration/test_hnsw_psql.py:172]: Random thought: We are now able to do `executor.status` or `executor.index` directly on the Executor object. I wonder, whether we should use the Executor object itself  whenever we are not explicitly use `shards` or `replicas`!? Would make an easier to understand test with much less boilercode-testing.

Comment (cristianmtr) [reply tests/integration/test_hnsw_psql.py:172]: Hm? Those would be unit tests, right? There are difference when in a Flow, like parameters and returns being handled automatically. And there are already unit tests directly on the class in the tests/unit folder

File: README.md (hunk 2 above)
Comment (cristianmtr) [reply README.md]: done

**Review by numb3r3 — APPROVED** [review]

**Pull request merged**